### PR TITLE
Change ics parser

### PR DESCRIPTION
**Pull request by mlsteele**

### CODE REVIEW

File: go/chat/wallet/parser_test.go [hunk 1]
```diff
@@ -52,26 +63,38 @@ func TestFindCandidates(t *testing.T) {
 		candidateTestCase{"thanks friend,+25eur", []ChatTxCandidate{anonTx}},
 		candidateTestCase{"thanks friend,+25eur\nnewline", []ChatTxCandidate{anonTx}},
 		candidateTestCase{"thanks friend,+25eur\ttabbed", []ChatTxCandidate{anonTx}},
+		candidateTestCase{"thanks friend (+25eur)\ttabbed", []ChatTxCandidate{anonTx}},
```
Comment (heronhaye) [inline go/chat/wallet/parser_test.go]: great

File: go/chat/wallet/parser.go [hunk 2]
```diff
@@ -9,8 +9,10 @@ import (
 )
 
 var txPattern = regexp.MustCompile(
-	// Must explicitly have a + in front
-	`\+` +
+	// Start at the beginng of the line, space, or some hand picked artisanal characters
```
Comment (heronhaye) [inline go/chat/wallet/parser.go]:
```suggestion
	// Start at the beginning of the line, space, or some hand picked artisanal characters
```

**Comment by mlsteele** [comment]
Changed it again. Because startIndex was ending up too early. Now it uses a capture group to decide where the `+` is. Had to change a decorate test.

File: go/chat/wallet/parser.go [hunk 3]
```diff
@@ -40,17 +42,26 @@ func FindChatTxCandidates(xs string) []ChatTxCandidate {
 	allRawIndices := txPattern.FindAllStringSubmatchIndex(replaced, maxTxsPerMessage)
 	matches := make([]ChatTxCandidate, 0, len(allRawIndices))
 	for _, rawIndices := range allRawIndices {
-		amount := xs[rawIndices[2]:rawIndices[3]]
+		group := func(n int) (s string, startIndex, endIndex int) {
+			startIndex, endIndex = rawIndices[2*n], rawIndices[2*n+1]
+			if startIndex >= 0 {
```
Comment (heronhaye) [inline go/chat/wallet/parser.go]: what is this for? when startIndex -1?

Comment (mlsteele) [reply go/chat/wallet/parser.go]: yes that

**Pull request merged**